src/lib/deploy.js

```diff
@@ -40,6 +41,30 @@ async function deploy({ alias, yes }) {
     return;
   }
 
+  // Query npm registry to get the latest CLI version.
+  const latestCliVersion = await fetch(
```

Comment:
nit: there's a lot of logic now related to finding the current / latest version, amidst the deploy logic. It would make sense to me to factor this out into separate functions, to reduce spaghetti and make it usable for other commands in the future

maybe something like:
```js
const latestCliVersion = await getLatestVersion();
const installedCliVersion = await getInstalledVersion();
if (hasBreakingChanges(installedCliVersion, latestCliVersion)) {
...
```